misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -298,7 +305,8 @@ private CoverageBuilder analyzeStructure() throws IOException {
         return coverageBuilder;
     }
 
-    private void createReport(final IBundleCoverage bundleCoverage, Map<String, ModuleCoverage> moduleCoverageMap) {
+    private void createReport(final IBundleCoverage bundleCoverage, Map<String, ModuleCoverage> moduleCoverageMap,
```

Comment:
There should be a logic to skip adding the generated files here. The generated files will be in the exclusion list mentioned here https://github.com/ballerina-platform/ballerina-lang/pull/36239/files#r893189786

Reply:
Implemented an exclusion list that takes document Ids. Needs to be reviewed